### ISSUE TITLE
Having troubles making NON_SUBSCRIPTION/NON_CONSUMABLE purchases on iOS

### ISSUE THREAD

**Issue by mcfarljw** [RevenueCat/purchases-capacitor#106]
I'm trying to switch over from the https://github.com/revenuecat/cordova-plugin-purchases plugin but I'm running into issues with purchasing NON_SUBSCRIPTION/NON_CONSUMABLE products. I haven't had any issues with purchasing recurring SUBSCRIPTION type products.

Here is an example of one of the packages:

```json
{
  "product": {
    "price": 6.99,
    "discounts": [],
    "introPrice": null,
    "priceString": "$6.99",
    "identifier": "[redacted]",
    "description": "Unlocks all content, features and removes ads",
    "productCategory": "NON_SUBSCRIPTION",
    "productType": "NON_CONSUMABLE",
    "title": "Full Unlock",
    "subscriptionPeriod": null,
    "currencyCode": "USD"
  },
  "offeringIdentifier": "full_unlock",
  "identifier": "$rc_lifetime",
  "packageType": "LIFETIME"
}
```

I've tried both `purchasePackage` and `purchaseStoreProduct`. An error isn't thrown and the promise never completes. The purchase popup on iOS is never shown. Here is what I've tried:

```ts
await Purchases.purchasePackage({ aPackage: this.package })
await Purchases.purchaseStoreProduct({ product: this.package.product })
```

In the Cordova plugin I had to specifically set `PURCHASE_TYPE.INAPP` when purchasing these kinds of non-consumable products but this seems to no longer be an option.

```ts
// https://revenuecat.github.io/cordova-plugin-purchases-docs/4.7.1/classes/default.html#purchaseProduct
await Purchases.purchaseProduct(this.package.product.identifier, null, PURCHASE_TYPE.INAPP)
```

### Other Notes

I noticed this debug warning in the logs but I'm not sure if it's related to this issue or not.

```
2023-09-08 17:12:55.031426-0400 App[920:161684] [store_kit] DEBUG: ℹ️ This StoreProduct represents an SK1 product, the type of product cannot be determined, the value will be undefined. Use `StoreProduct.productCategory` instead.
```

**Comment by tonidero** [comment]
Hi @mcfarljw, unfortunately, I haven't been able to reproduce the issue. I was able to purchase a non-consumable product in iOS in our tester app using both `purchasePackage` and `purchaseStoreProduct`. Your code sample seems to be correct, I'm guessing you're using the `Purchases.getOfferings` method to get the package correct? 

In order to help us diagnose the issue, could you send full debug logs from configuring the SDK to the moment you call `purchasePackage`/`purchaseStoreProduct`? 

If you prefer, feel free to open a ticket through the RC dashboard referencing this issue to send the logs that way.

**Comment by mcfarljw** [comment]
Correct, I am using `Purchases.getOfferings` to get the the package/product from the offering, then directly passing that response when trying to make the purchase. I tried with build deployed to TestFlight and then also ones directly pushed to my device from Xcode. Not sure if it makes a difference but all our products have all been approved for production by Apple.

Just to be thorough I also tried installing `cordova-plugin-purchases` again and it worked as expected. Here are the dependencies:

```
"@awesome-cordova-plugins/core": "^6.4.0",
"@awesome-cordova-plugins/purchases": "^6.4.0",
"cordova-annotated-plugin-android": "^1.0.4",
"cordova-plugin-purchases": "^4.7.1",
```

I'll get lots passed your way soon. I appreciate you taking the time to try out a non-consumable test purchase on your end. I was hoping it was something that was going to be a quick repro.

**Comment by mcfarljw** [comment]
I have resolved the issue. It was an issue with getting a reactive product from a Pinia store using Vue. The solution was to use `toRaw` (https://vuejs.org/api/reactivity-advanced.html#toraw). No errors were thrown but I noticed that setting a breakpoint on `purchaseStoreProduct` was never getting called on iOS, oddly everything worked on Android so maybe it just handled the reactive object elegantly.

```ts
const { customerInfo } = await Purchases.purchaseStoreProduct({
  product: toRaw(this.product),
})
```
